api/src/main/java/org/openmrs/api/storage/S3/S3StorageService.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.api.storage.S3;
+
+import java.util.UUID;
+
+
+import org.openmrs.api.StorageService;
+import org.openmrs.api.storage.BaseStorageService;
+import org.openmrs.api.storage.ObjectMetadata;
+import org.openmrs.api.storage.StorageServiceCondition;
+import org.openmrs.api.stream.StreamDataService;
+import org.openmrs.api.stream.StreamDataWriter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.stereotype.Service;
+
+import java.io.*;
+import java.util.stream.Stream;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.ListObjectsV2Request;
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+
+
+
+
+
+@Service
+@Qualifier("s3")
+@Conditional(StorageServiceCondition.class)
+public class S3StorageService extends BaseStorageService implements StorageService {
+
+    protected AmazonS3 s3Client;
+    private final String bucketName = "PLACEHOLDER_BUCKET_NAME"; // Replace with your actual S3 bucket name
+
+    @Autowired
+public S3StorageService(StreamDataService streamDataService) {
+    super(streamDataService);
+    this.s3Client = AmazonS3ClientBuilder.standard().build();
```

Comment:
Did you check if it is thread-safe? Can it be re-used by multiple threads?

Reply:
Can we have it configured with credentials from properties injected via @Value and only if missing use standard configuration? The reason is that we will have more ways to provide these credentials via e.g. openmrs-runtime.properties. Please use storage.s3.secretAccessKey, storage.s3.accessKeyId, storage.s3.region.